transcom/mymove
# B 19845 int

## [B-19845](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19845)

## Summary

This story creates UI elements during the customer's expense reporting process that capture additional sit data, namely the location and storage weight that is included on their receipt. These fields are required when adding a storage expense type on any moving expense. These fields have been added and frontend validation for all fields has been corrected, where there were previously some bugs (read about this more in the testing instructions below). 

> [!TIP]
> This story is downstream of [B-20173](https://github.com/transcom/mymove/pull/12825), and contains it's changes. However, this PR may be easier to review once B-20173 has been merged into integrationTesting.

This story has taken slightly longer to complete due to the issue that B-20173 resolves and the incorrectly working existing frontend validation system, however has still been completed within original estimate time.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Access the customer app, create a new user, create and submit a PPM. Details aren't very important here
2. Access the office app, login as a SC, approve the PPM.
3. Access the customer app as your original user, click the add ppm documents button.
4. Continue through the Trip page using dummy data to get to the Expenses page.
5. Once on the expenses page, ensure the UI matches what is outlined in the story's AC when a storage type is selected. It should look something like this: 
<img width="759" alt="image" src="https://github.com/transcom/mymove/assets/148906585/7929b4b6-6b0e-4ac5-824a-991b4a2aeb81">

6. Ensure that the new sit-related fields are required, but only when the type of expense is storage.
7. Submit the expense ticket with all information, and to confirm that it has saved, edit the same expense ticket after to see your data in the input fields.
8. Edit the type of expense ticket to something other than storage, and save it. Ensure it saves properly, and the now-invisible storage fields are not required.
9. Edit the ticket again, and change the type to storage. Note how the storage-related fields are now empty, as the ticket was changed to a type that does not use those fields.

> [!IMPORTANT]  
> Until this ticket, sit-related fields still had frontend validation applying from storage fields even when the type of ticket was not storage, due to a mistaken part of the validation schema. This has been fixed in this ticket, please test it rigorously. 
 

### Frontend
